File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    var head *ListNode
	var tmp *ListNode
	
	for list1 != nil && list2 != nil {
		var tmp2 *ListNode
		if list1.Val < list2.Val {
			tmp2 = list1
			list1 = list1.Next
		}else{
			tmp2 = list2
			list2 = list2.Next
		}
		
		if head == nil {
			head = tmp2
			tmp = tmp2
			continue
		}

		tmp.Next = tmp2
		tmp = tmp.Next
		// fmt.Println(head.Val,tmp.Val,tmp2.Val)
	}


	for list1 != nil {
		if head == nil {
			head = list1
			tmp = list1
		
		}else{
			tmp.Next = list1
			tmp = tmp.Next
		}
		
		list1 = list1.Next
	}
	
	for list2 != nil {
		if head == nil {
			head = list2
			tmp = list2
		
		}else{
			tmp.Next = list2
			tmp = tmp.Next
		}
		list2 = list2.Next
	}

	return head
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if not head.next:
            return None


        curr = head
        size = 0 
        
        while curr:
            size+=1
            curr = curr.next


        indexFromStart = size - n + 1


        # removing head
        if indexFromStart == 1:
            tmp = head.next
            head.next = None
            head = tmp

        
        # any other postion
        else:
            
            prev = head
            curr = head.next
            
            for i in range(2,size + 1):
            
                if indexFromStart == i:
                    prev.next = curr.next
                    curr.next = None
                    break

                prev = curr
                curr = curr.next




        return head